Stratis.Bitcoin/BlockPulling/QualityScore.cs

```diff
@@ -103,29 +92,28 @@ public void AddSample(IBlockPullerBehavior peer, long blockDownloadTimeMs, int b
 
             lock (this.lockObject)
             {
-                // If we reached the maximum number of samples, we need to remove oldest sample.
-                if (this.samplesCount == this.samples.Length)
-                {
-                    PeerSample oldSample = this.samples[this.samplesIndex];
+                // Add new sample to the mix.
+                PeerSample newSample = new PeerSample();
```

Comment:
if it is struct, which it is and this code is responsible for that, then allocation here means just extra 2 items on stack, i.e. this is safely in align with what was the original purpose